src/pylhe/__init__.py

```diff
@@ -120,14 +123,34 @@ def loads():
     pass
 
 
-def readLHEInit(thefile):
+def _open_gzip_file(filepath):
```

Comment:
this should probably rreturn a consistent type.. i.e. also a file object for the non-gz case.. i.e. `return open(filepath)`

if we're checkig the magic bytes anyways do we need the `gz` extension check?

Reply:
> this should probably return a consistent type.. i.e. also a file object for the non-gz case.. i.e. `return open(filepath)`

Yeah, that's a good point, though that will need some refactoring with current use, as it will currently fail if you give it a non-gzip-ed LHE file then as `ET.iterparse` tries to open up the file you give it.

> if we're checking the magic bytes anyways do we need the `gz` extension check?

This was just so files that don't match the naming convention get passed through to `ET.iterparse` to be opened up. If you do match, then we check if you're actually a `gz` file.

Reply:
So let me think on a refactor for that tonight and you can review again tomorrow. :+1:

Reply:
c.f. my ranting below. :P